frontend/src/components/maps/hooks/useActiveFeatureLayer.tsx

```diff
@@ -55,7 +61,14 @@ const useActiveFeatureLayer = ({
     let title = 'Municipality Information';
     let feature: Feature | undefined = undefined;
 
-    const parcel = await parcelsService.findOneWhereContains(latLng);
+    // call these APIs in parallel - notice there is no "await"
+    const task1 = parcelsService.findOneWhereContains(latLng);
+    const task2 = regionService.findMetadataByLocation(latLng, 'GEOMETRY');
+    const task3 = districtService.findMetadataByLocation(latLng, 'GEOMETRY');
```

Comment:
nit: could be retrieve parcelTask, retrieveRegionTask ...

Reply:
agree - but felt that they are close together so task1..taskn was okay